### ISSUE TITLE
Add a shitload of tests to rustdoc

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/rust#42018]
Since we now have a more complete way to compare two HTML contents, I think we can now _**strongly**_ reinforce rustdoc's tests (which are a bit waek and lacking for now).

cc @rust-lang/docs

**Comment by QuietMisdreavus** [comment]
Some ideas:

* `src/librustdoc/html/markdown.rs` already has a `#[cfg(test)]` block, but it doesn't test the "hide lines of code with `#` " behavior.
* `src/librustdoc/test.rs` could use some tests to verify how it adds attributes to the test, whether it wraps tests in a `fn main()`, things like that.

**Comment by QuietMisdreavus** [comment]
One thing i'd like to see (that will probably need cooperation from some rustbuild experts) is the ability to test the stdout-printing behavior of rustdoc in various situations. We don't have UI tests for rustdoc the same way the compiler does, so right now there's no place for these in the testing setup.

**Comment by camelid** [comment]
ping: Is this still an issue?

**Comment by GuillaumeGomez** [comment]
Still waiting on https://github.com/rust-lang/rust/pull/70533.

**Comment by rylev** [comment]
Triage: #70533 was superseded by https://github.com/rust-lang/rust/pull/79979 which has been merged. @GuillaumeGomez can this be closed? If not, we should probably add more detail to what kind of tests need to be added since this issue is not very actionable right now.

**Comment by GuillaumeGomez** [comment]
Still far from done unfortunately. Please keep it open.
